src/mrb_maxminddb.c

```diff
@@ -116,6 +116,9 @@ static mrb_value mrb_maxminddb_region(mrb_state *mrb, mrb_value self) {
   int status =
       MMDB_aget_value(&(data->lookup_result_s.entry), &entry_data, path);
 
+  if (MMDB_LOOKUP_PATH_DOES_NOT_MATCH_DATA_ERROR == status)
```

Comment:
`MMDB_LOOKUP_PATH_DOES_NOT_MATCH_DATA_ERROR` when the status is not the expected value from `MMDB_aget_value`

https://github.com/maxmind/libmaxminddb/blob/07412b30f204034bfec880a38763a222d7d36043/src/maxminddb.c#L1241

```c
int MMDB_aget_value(MMDB_entry_s *const start,
                    MMDB_entry_data_s *const entry_data,
                    const char *const *const path)
{

(snip)

while (NULL != (path_elem = path[i++])) {
    DEBUG_NL;
    DEBUG_MSGF("path elem = %s", path_elem);

    /* XXX - it'd be good to find a quicker way to skip through these
       entries that doesn't involve decoding them
       completely. Basically we need to just use the size from the
       control byte to advance our pointer rather than calling
       decode_one(). */
    if (entry_data->type == MMDB_DATA_TYPE_ARRAY) {
        int status = lookup_path_in_array(path_elem, mmdb, entry_data);
        if (MMDB_SUCCESS != status) {
            memset(entry_data, 0, sizeof(MMDB_entry_data_s));
            return status;
        }
    } else if (entry_data->type == MMDB_DATA_TYPE_MAP) {
        int status = lookup_path_in_map(path_elem, mmdb, entry_data);
        if (MMDB_SUCCESS != status) {
            memset(entry_data, 0, sizeof(MMDB_entry_data_s));
            return status;
        }
    } else {
        /* Once we make the code traverse maps & arrays without calling
         * decode_one() we can get rid of this. */
        memset(entry_data, 0, sizeof(MMDB_entry_data_s));
        return MMDB_LOOKUP_PATH_DOES_NOT_MATCH_DATA_ERROR;
    }
}

return MMDB_SUCCESS;
}
```